apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/FileUpload.kt

```diff
@@ -1,5 +1,19 @@
 package com.apollographql.apollo.api
 
-expect class FileUpload constructor(mimetype: String, file: File) {
-  val mimetype: String
+class FileUpload(val mimetype: String, val filePath: String) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is FileUpload) return false
+
+    if (mimetype != other.mimetype) return false
+    if (filePath != other.filePath) return false
+
+    return true
+  }
+
+  override fun hashCode(): Int {
+    var result = mimetype.hashCode()
+    result = 31 * result + filePath.hashCode()
```

Comment:
Will Kotlin not auto-generate `hashCode` for MPP stuff?

Reply:
What do you mean? It won't as `FileUpload` is not a `data` class.

Reply:
Why is it not? 😊

Reply:
Ah that's what I missed, that it's not a `data class`. However @tasomaniac poses a good question 😄

Reply:
I'd like to avoid data classes in public APIs unless really needed as they make it quite hard to ensure source/binary compatibility.
See https://jakewharton.com/public-api-challenges-in-kotlin/

Reply:
And that's a great answer as to why not do it that way. Thank you @martinbonnin!

Reply:
Oh yes. Makes sense.